sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStoreChangelog.scala

```diff
@@ -102,6 +115,13 @@ abstract class StateStoreChangelogWriter(
     fm.createAtomic(file, overwriteIfPossible = true)
   protected var compressedStream: DataOutputStream = compressStream(backingFileStream)
 
+  def writeLineage(lineage: Array[LineageItem]): Unit = {
```

Comment:
Is there anything that guarantees that this is going to be called first so that it goes into the first line?